README.md

```diff
@@ -88,8 +91,46 @@ To set your API credentials for an API request,Configure the following informati
     keyFileName         = your_portfolio_id
     keysDirectory       = Resource
     useMetaKey          = true
+    enableClientCert    = false
 ```
 
+
+   * OAuth
+   
+   To use OAuth, please follow the steps. Client Certificate is required to hit OAuth API. If the certificate (Public Key) and Private Key are in 2 different files, merge them into a single .p12 file using openssl. 
+
+```
+   openssl pkcs12 -export -out certificate.p12 -inkey privateKey.key -in certificate.crt
+```
+
+   Set the config below
+
+```java
+   authenticationType  = MutualAuth
+   enableClientCert    = true
+   clientCertDirectory = resources
+   clientCertFile      = your_client_cert - .p12 format
+   clientCertPassword  = password_for_client_cert
+   clientId            = your_client_id
+   clientSecret        = your_client_secret
+```
+
+   Hit OAuth API with the above configuration, to get the Access Token and Refresh Token. Once the tokens are obtained, set the config below
+
+```java
+   authenticationType  = OAuth
+   enableClientCert    = true
+   clientCertDirectory = resources
+   clientCertFile      = your_client_cert - .p12 format
+   clientCertPassword  = password_for_client_cert
+   clientId            = your_client_id
+   clientSecret        = your_client_secret
+   accessToken         = generated_access_token
+   refreshToken        = generated_refresh_token
+```
+
+   With the above config, we can use OAuth to hit other APIs. Once the Access Token expires, use the Refresh Token to generate another Access Token.   
+
```

Comment:
* When do I need to use MutualAuth and when do I need to use OAuth?

* How often do I need to use MutualAuth flow? 
* How long are the tokens active? This means, how long can I store the tokens before I start using them?
* When do the tokens expire? This means, how long can I keep using the same token?

* What errors are possible? 
* What happens when an error is encountered? 
* If I get an error, is the AuthCode OR RefreshToken OR AccessToken still valid?

* Which environment will support OAuth flows? 
* Can I try it in PROD? 
* Which APIs will accept OAuth tokens?